### PR TITLE
mkfs-util: Silence noisy warning from mkfs.btrfs

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

File: src/shared/mkfs-util.c [hunk 1]
```diff
@@ -388,6 +388,11 @@ int make_filesystem(
                 if (quiet && strv_extend(&argv, "-q") < 0)
                         return log_oom();
 
+                /* mkfs.btrfs unconditionally warns about several settings changing from v5.15 onwards which
+                 * isn't silenced by "-q", so let's redirect stdout to /dev/null as well. */
+                if (quiet)
+                        stdio_fds[1] = -EBADF;
```
Comment (poettering) [inline src/shared/mkfs-util.c:394]: hmm, wouldn't it be nicer to just add arguments to our invocation that match the defaults on v5.15 onwards? that we we get same behaviour everywhere, and no warnings.

Comment (daandemeyer) [reply src/shared/mkfs-util.c:394]: If only it were that easy, the warning is printed unconditionally, regardless of what settings we use

Comment (poettering) [reply src/shared/mkfs-util.c:394]: what? that's stupid... they warn about issues the mkfs invocation is not actually suffering by? hmpf

Comment (poettering) [reply src/shared/mkfs-util.c:394]: hmm, i wonder if we really should do this though, isn't this taping over some fuckup in mkfs.btrfs?

at the very least they should make -q silence this

Comment (bluca) [reply src/shared/mkfs-util.c:394]: if they fix it we can revise later, this is annoying to deal with otherwise

**Pull request merged**